source/configuration/modules/omfile.rst

```diff
@@ -1,434 +1,437 @@
 
-The omfile plug-in provides the core functionality of writing messages
-to files residing inside the local file system (which may actually be
-remote if methods like NFS are used). Both files named with static names
-as well files with names based on message content are supported by this
-module.
+The omfile plugin provides the needed functionality to write messages to a local file.
 
 Configuration Parameters
-------------------------
-Omfile is a built-in module that does not need to be loaded. In order to
-specify module parameters, use
+************************
 
-::
-
-   module(load="builtin:omfile" ...parameters...)
-
-Note that legacy directives **do not** affect new-style RainerScript configuration
-objects. See :doc:`basic configuration structure doc <../basic_structure>` to
-learn about different configuration languages in use by rsyslog.
+Omfile is a built-in module that does not need to be loaded for default settings.
+In order to specify module parameters, it must be loaded using:
 
-General Notes
-^^^^^^^^^^^^^
+    module(load="builtin:omfile" [parameters])
 
 As can be seen in the parameters below, owner and groups can be set either by
-name or by direct id (uid, gid). While using a name is more convenient, using
+name or by id (*uid, gid*). While using a name is more convenient, using
 the id is more robust. There may be some situations where the OS is not able
-to do the name-to-id resolution, and these cases the owner information will be
-set to the process default. This seems to be uncommon and depends on the
+to do the name-to-id resolution and -in these cases- the owner information will be
+set to defaults. This seems to be uncommon and depends on the
 authentication provider and service start order. In general, using names
 is fine.
 
-Module Parameters
-^^^^^^^^^^^^^^^^^
-
-.. function::  template [templateName]
-
-   *Default: RSYSLOG_FileFormat*
-
-   Set the default template to be used if an action is not configured
-   to use a specific template.
-
-.. function::  dirCreateMode [octalNumber]
-
-   *Default: 0700*
-
-   Sets the default dirCreateMode to be used for an action if no
-   explicit one is specified.
-
-.. function::  fileCreateMode [default 0644] [octalNumber]
-
-   *Default: 0644*
+**TODO according to @davidelang this is insecure https://github.com/rsyslog/rsyslog-doc/pull/294#pullrequestreview-14280545**
+Also, consider *octalNumber* values must always be a 4-digit octal number, with the initial digit being zero.
+Mind that actual permission depend on rsyslogd's process umask. If in doubt, use "$umask 0000" right at the beginning of the
+configuration file to remove any restrictions.
 
-   Sets the default fileCreateMode to be used for an action if no
-   explicit one is specified.
-
-.. function:: fileOwner [userName]
-
-   *Default: process user*
-
-   Sets the default fileOwner to be used for an action if no
-   explicit one is specified.
+Module Parameters
+=================
```

Comment:
I would consider something along the lines of a table listing the option names and then a  link pointing at the same parameters that can be set in the action.